CMakeLists.txt

```diff
@@ -168,18 +165,21 @@ function(set_compiler_flags target cpp_standard target_arch)
   target_compile_options(${target} PRIVATE "$<$<CXX_COMPILER_ID:GNU>:-fno-builtin-memcpy>")
   target_compile_options(${target} PRIVATE "$<$<CXX_COMPILER_ID:GNU>:-fno-builtin-memset>")
 
-  # Check for ${target_arch} and set it or use "march=native" if not defined
+  # Check for ${target_arch} and set it or use the current system if not defined
   if("${target_arch}" STREQUAL "")
-    # MSVC does not have a direct equivalent to -march=native
-    target_compile_options(
-      ${target} PRIVATE
-      "$<$<AND:$<CXX_COMPILER_ID:GNU>,$<NOT:$<CXX_COMPILER_ID:AppleClang>>>:-march=native>"
-      "$<$<CXX_COMPILER_ID:MSVC>:/arch:AVX2>")
+    # Only use the current system if we are not cross compiling
+    if((NOT CMAKE_CROSSCOMPILING) OR (CMAKE_SYSTEM_PROCESSOR MATCHES CMAKE_HOST_SYSTEM_PROCESSOR))
+      # MSVC does not have a direct equivalent to -march=native
+      target_compile_options(
+        ${target} PRIVATE
+        "$<$<CXX_COMPILER_ID:GNU,Clang,AppleClang>:-march=native>"
```

Comment:
I believe `AppleClang` doesn't support `-march=native`. Have you tried running it on MacOS?

Reply:
Honestly I'm not sure of the actual support of that flag. But the MacOs action builds, so it is at least supported for MacOs x86_64.
However it may not be supported on ARM processors (e.g. M1) although it might be on newer versions.

Maybe we need some more actions to build for ARM targets properly (rather than cross compiling).

Reply:
> I believe `AppleClang` doesn't support `-march=native`. Have you tried running it on MacOS? 

Well whatever `AppleClang` supports, I've updated it now to only use the `march=native` flag if the compiler supports it.